web/ui/react-app/src/pages/graph/PanelList.tsx

```diff
@@ -1,219 +1,178 @@
+  const query = encodePanelOptionsToQueryString(nextPanels);
+  window.history.pushState({}, '', query);
+};
+
+interface PanelListProps extends PathPrefixProps, RouteComponentProps {
   panels: PanelMeta[];
-  pastQueries: string[];
-  metricNames: string[];
-  fetchMetricsError: string | null;
-  timeDriftError: string | null;
+  metrics: string[];
   useLocalTime: boolean;
+  queryHistoryEnabled: boolean;
 }
 
-class PanelList extends Component<RouteComponentProps & PathPrefixProps, PanelListState> {
-  constructor(props: RouteComponentProps & PathPrefixProps) {
-    super(props);
-
-    this.state = {
-      panels: decodePanelOptionsFromQueryString(window.location.search),
-      pastQueries: [],
-      metricNames: [],
-      fetchMetricsError: null,
-      timeDriftError: null,
-      useLocalTime: this.useLocalTime(),
-    };
-  }
-
-  componentDidMount() {
-    !this.state.panels.length && this.addPanel();
-    fetch(`${this.props.pathPrefix}/api/v1/label/__name__/values`, { cache: 'no-store', credentials: 'same-origin' })
-      .then(resp => {
-        if (resp.ok) {
-          return resp.json();
-        } else {
-          throw new Error('Unexpected response status when fetching metric names: ' + resp.statusText); // TODO extract error
-        }
-      })
-      .then(json => {
-        this.setState({ metricNames: json.data });
-      })
-      .catch(error => this.setState({ fetchMetricsError: error.message }));
-
-    const browserTime = new Date().getTime() / 1000;
-    fetch(`${this.props.pathPrefix}/api/v1/query?query=time()`, { cache: 'no-store', credentials: 'same-origin' })
-      .then(resp => {
-        if (resp.ok) {
-          return resp.json();
-        } else {
-          throw new Error('Unexpected response status when fetching metric names: ' + resp.statusText); // TODO extract error
-        }
-      })
-      .then(json => {
-        const serverTime = json.data.result[0];
-        const delta = Math.abs(browserTime - serverTime);
-
-        if (delta >= 30) {
-          throw new Error(
-            'Detected ' +
-              delta +
-              ' seconds time difference between your browser and the server. Prometheus relies on accurate time and time drift might cause unexpected query results.'
-          );
-        }
-      })
-      .catch(error => this.setState({ timeDriftError: error.message }));
-
+export const PanelListContent: FC<PanelListProps> = ({
+  metrics = [],
+  useLocalTime,
+  pathPrefix,
+  queryHistoryEnabled,
+  ...rest
+}) => {
+  const [panels, setPanels] = useState(rest.panels);
+  const [historyItems, setLocalStorageHistoryItems] = useLocalStorage<string[]>('history', []);
+
+  useEffect(() => {
+    !panels.length && addPanel();
     window.onpopstate = () => {
       const panels = decodePanelOptionsFromQueryString(window.location.search);
       if (panels.length > 0) {
-        this.setState({ panels });
+        setPanels(panels);
```

Comment:
Somehow panel options aren't loaded correctly from the query string anymore.

Try `https://deploy-preview-6629--prometheus-react.netlify.com/new/graph?g0.expr=irate(node_cpu_seconds_total%5B1m%5D)&g0.tab=0&g0.stacked=1&g0.range_input=1m&g1.expr=irate(node_cpu_seconds_total%5B1m%5D)&g1.tab=1&g1.stacked=0&g1.range_input=1h`

...it will just change itself to `https://deploy-preview-6629--prometheus-react.netlify.com/new/graph?g0.expr=&g0.tab=1&g0.stacked=0&g0.range_input=1h` and not show any panels.